concordium-node/src/grpc2.rs

```diff
@@ -610,6 +617,102 @@ pub mod server {
             Ok(response)
         }
 
+        async fn get_instance_state(
+            &self,
+            request: tonic::Request<crate::grpc2::types::InstanceInfoRequest>,
+        ) -> Result<tonic::Response<Self::GetInstanceStateStream>, tonic::Status> {
+            let request = request.get_ref();
+            let block_hash = request.block_hash.as_ref().require()?;
+            let contract_address = request.address.as_ref().require()?;
+            let (hash, response) =
+                self.consensus.get_instance_state_v2(block_hash, contract_address)?;
+            match response {
+                ContractStateResponse::V0 {
+                    state,
+                } => {
+                    // We need to return the same type in both branches, so we create a silly
+                    // little channel to which we will only send one value.
+                    let (sender, receiver) = tokio::sync::mpsc::channel(1);
+                    let _sender = tokio::spawn(async move {
+                        let msg = types::InstanceStateKvPair {
+                            key:   Vec::new(),
+                            value: state,
+                        };
+                        let _ = sender.send(Ok(msg)).await;
+                        // The error only happens if the receiver has been
+                        // dropped already (e.g., connection closed),
+                        // so we do not have to handle it. We just stop
+                    });
+                    let mut response =
+                        tonic::Response::new(tokio_stream::wrappers::ReceiverStream::new(receiver));
+                    add_hash(&mut response, hash)?;
+                    Ok(response)
+                }
+                ContractStateResponse::V1 {
+                    state,
+                    mut loader,
+                } => {
+                    let (sender, receiver) = tokio::sync::mpsc::channel(10);
```

Comment:
Why was 10 chosen?

Reply:
I added a comment now. Because it's a big small number and provides some buffer, but not too much so that we'd have to keep too much data in memory in case the consumer is slow.